docs/source/use_cases/run_picard_and_gatk/index.rst

```diff
@@ -17,20 +17,178 @@ __ RenderedVersion_
+
+3. Build Picard version with GA4GH support: 
+
+.. code-block:: shell
+
+  $ cd picard
+  $ ant -lib lib/ant package-commands-ga4gh
+
+4. Make sure you put **client_secrets.json** file in the parent folder just above Picard.
+You should end up with the following directory structure:
+
+.. code-block:: shell
+
+  your_client_folder \
+    client_secrets.json
+    gatk-tools-java \
+    picard \
+      htsjdk \
+
+
+Run Picard tools with an input from the cloud
+=============================================
+You can now run ViewSam tool that prints the contents of the supplied INPUT 
+
+.. code-block:: shell
+
+  $ cd picard
+  $ java \
+     -jar dist/picard.jar ViewSam \
+     INPUT=https://www.googleapis.com/genomics/v1beta2/readgroupsets/CK256frpGBD44IWHwLP22R4/ \
+     GA4GH_CLIENT_SECRETS=../client_secrets.json
+
+This command uses an older, slower REST based API. To run using GRPC API implementation (which is much faster) use the following command that utilizes ALPN jars that come with gatk-tools-java and enables GRPC support:
+
+.. code-block:: shell
+
+  java \
+   -Xbootclasspath/p:../gatk-tools-java/lib/alpn-boot-8.1.3.v20150130.jar \
+   -Dga4gh.using_grpc=true \
+   -jar dist/picard.jar ViewSam \
+   INPUT=https://www.googleapis.com/genomics/v1beta2/readgroupsets/CK256frpGBD44IWHwLP22R4/ \
+   GA4GH_CLIENT_SECRETS=../client_secrets.json
+
+For Java 7 (as opposed to 8) use *alpn-boot-7.1.3.v20150130.jar*.
+
+We use a test readset here from `genomics-test-data <https://pantheon.corp.google.com/project/genomics-test-data/storage/browser/gatk-tools-java/>`_ project. 
+
+Specifying a genomics region to use from the readset
+----------------------------------------------------
+The INPUT urls are of the form **https://<GA4GH provider>/readgroupsets/<readgroupset id>[/sequence][/start-end]**.
+
+For example:
+
+.. code-block:: shell
+
+  java -jar dist/picard.jar ViewSam \ 
+  INPUT=https://www.googleapis.com/genomics/v1beta2/readgroupsets\
+  /CMvnhpKTFhD3he72j4KZuyc/chr17/41196311-42677499 \
+  GA4GH_CLIENT_SECRETS=../client_secrets.json
+
+
+Timing the reading speed from the cloud
+---------------------------------------
+You can run `gatk-tools-java/src/main/scripts/example.sh <https://github.com/googlegenomics/gatk-tools-java/blob/master/src/main/scripts/example.sh>`_ with and without "grpc" command line parameter to see the difference in reading speed. The timing statistics are dumped to the terminal.
+We benchmarked **x11** speed improvements with GRPC compared to REST, giving **~12,000 reads/second**.
+
+The tests were done on `Platinum Genomes NA12877_S1.bam dataset <https://pantheon.corp.google.com/storage/browser/genomics-public-data/platinum-genomes/bam/?_ga=1.197206447.160385476.1431305548>`_, please see the `detailed writeup of the test procedure and results <https://docs.google.com/document/d/1Br7RMSbAChNpG6pi2teujf-YthczF-rAM1afSgDoCgQ/edit#>`_ if you want to repeat the test.
```

Comment:
https://console.developers.google.com instead of https://pantheon.corp.google.com